tools/lkl/Makefile.autoconf

```diff
@@ -101,6 +106,8 @@ define do_autoconf
   $(eval CC := $(CROSS_COMPILE)gcc)
   $(eval LD_FMT := $(shell $(LD) -r -print-output-format))
   $(eval EXEC_FMT := $(shell echo $(LD_FMT) | cut -d "-" -f1))
+  $(if $(or $(filter $(EXEC_FMT),elf64),$(filter $(LD_FMT),pe-x86-64)),$(call set_kernel_config,LKL_CONFIG_64BIT,1))
+  $(if $(filter $(EXEC_FMT),elf64-s390),$(call set_kernel_config,LKL_CONFIG_ENDIAN,1))
   $(if $(filter $(EXEC_FMT),$(POSIX_HOSTS)),$(call posix_host,$(LD_FMT)))
```

Comment:
do we want to add "pe-x86-64" for LKL_CONFIG_64BIT target ?
because it's defined in our Kconfig file (we don't have a test for that though).

https://github.com/lkl/linux/blob/master/arch/lkl/Kconfig#L52

```
config 64BIT
       def_bool y if "$(EXEC_FORMAT)" = "elf64" || "$(OUTPUT_FORMAT)" = "pe-x86-64"
```

Reply:
Thanks for the review, I'll fix the commit messages and typos. pe-x86-64 should be covered by the first line `$(filter $(LD_FMT),pe-x86-64)`

I have also noticed that the PCI tests have failed, do you have any idea why?

Reply:
> Thanks for the review, I'll fix the commit messages and typos. pe-x86-64 should be covered by the first line `$(filter $(LD_FMT),pe-x86-64)`

you're right.  my eyes are still in holiday mode :(

> I have also noticed that the PCI tests have failed, do you have any idea why?

I have no clue at a glance on the CI failure.. I will look into detail.

We also have a failure on android arm32, which looks like due to changes to the upstream kernel to not support gold linker for the kernel build.  it would be on option to android-arm32 target if there is not much interest.

Reply:
Yeah, I am not sure what to do about the gold linker situation, we can either drop the checks or carry the revert to renewable gold.

For the PCI failures I see that most of the time in CI the emulator fails to start, so maybe these issues were introduced earlier? I'll try to repro this locally and get more info.

Reply:
Found the PCI issue, CONFIG_DMA_OPS is not enabled by default anymore. PCI tests now pass, I'll let CI run and then I will merge the PR.

Reply:
Great! Thanks for your time.